flink-python/pyflink/datastream/functions.py

```diff
@@ -324,3 +364,16 @@ def __init__(self, sink_func: Union[str, JavaObject]):
         :param sink_func: The java SinkFunction object or the full name of the SinkFunction class.
         """
         super(SinkFunction, self).__init__(sink_func)
+
+
+class PartitionCustomMapFunction(MapFunction):
```

Comment:
Add this function here would confuse users since this is not a public Function. I think we can find other ways to judge whether this is a  `PartitionCustomMapFunction`. For example, we can add a local `MapFunction` returning a special `__expr__`(toString).